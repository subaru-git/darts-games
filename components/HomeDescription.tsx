import React, { FC } from 'react'
import { Center, Grid, GridItem, Image, Text } from '@chakra-ui/react'
import useLocale from '../hooks/locale'

const HomeDescription: FC = () => {
  const { t } = useLocale()
  return (
    <Grid templateColumns='repeat(2, auto)' justifyContent='center' gap={4}>
      <GridItem>
        <Image src='/darts.jpg' alt='darts' maxWidth='700px' />
      </GridItem>
      <GridItem>
        <Center h='100%'>
          <Text fontSize='xl' whiteSpace='pre-wrap'>
            {t.HOME_DESCRIPTION}
          </Text>
        </Center>
      </GridItem>
    </Grid>
  )
}

export default HomeDescription
