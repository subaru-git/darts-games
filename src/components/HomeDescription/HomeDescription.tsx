import React, { FC } from 'react'
import { Center, Grid, GridItem, Img, Text, useBreakpointValue } from '@chakra-ui/react'
import useLocale from '@/hooks/locale'

const HomeDescription: FC = () => {
  const isMd = useBreakpointValue({ base: false, md: true })
  return <>{isMd ? <DesktopHomeDescription /> : <MobileHomeDescription />}</>
}

const DesktopHomeDescription: FC = () => {
  const { t } = useLocale()
  return (
    <Grid templateColumns='repeat(2, auto)' justifyContent='center' gap={4}>
      <GridItem>
        <Img src='/darts.png' alt='darts homes' boxSize={450} objectFit='cover' loading='lazy' />
      </GridItem>
      <GridItem>
        <Center h='100%'>
          <Text fontSize={'xl'} whiteSpace='pre-wrap'>
            {t.HOME_DESCRIPTION}
          </Text>
        </Center>
      </GridItem>
    </Grid>
  )
}

const MobileHomeDescription: FC = () => {
  const { t } = useLocale()
  return (
    <>
      <Center>
        <Img src='/darts.png' alt='darts homes' boxSize={200} objectFit='cover' loading='lazy' />
      </Center>
      <Center h='100%'>
        <Text fontSize={'md'} whiteSpace='pre-wrap'>
          {t.HOME_DESCRIPTION}
        </Text>
      </Center>
    </>
  )
}
export default HomeDescription
