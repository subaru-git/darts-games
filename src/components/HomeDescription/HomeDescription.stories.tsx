/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import HomeDescription from './HomeDescription';

export default {
  title: 'Components/HomeDescription',
  component: HomeDescription,
} as ComponentMeta<typeof HomeDescription>;

const Template: ComponentStory<typeof HomeDescription> = (args) => <HomeDescription {...args} />;

export const Default = Template.bind({});
Default.args = {};
