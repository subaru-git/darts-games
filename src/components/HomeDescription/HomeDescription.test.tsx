import { render, screen } from '@testing-library/react'
import HomeDescription from './HomeDescription'

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }))
test('should rendering', () => {
  const { container } = render(<HomeDescription />)
  expect(screen.getByAltText('darts')).not.toBeNull()
  expect(container).toMatchSnapshot()
})
