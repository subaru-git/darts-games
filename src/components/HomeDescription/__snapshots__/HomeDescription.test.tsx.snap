// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-vyd9jy"
  >
    <div
      class="css-0"
    >
      <img
        alt="darts homes"
        class="chakra-image css-1ymdu7n"
        src="/darts.webp"
      />
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-tkevr6"
      >
        <p
          class="chakra-text css-1ngz931"
        >
          This website was created with the goal of creating a game for practicing darts at home.
Games are created based on YouTube and more.
Also, this website is processed on the client side, so it is impossible to synchronize results between devices.
You can do this by importing and exporting results, or you can manage it by posting to social networking sites such as Twitter.
(You don't want to sign into this incomprehensible site, do you?)
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
